### PR TITLE
Move UI tests to Windows 11

### PR DESCRIPTION
**Pull request by mcooley**
Calculator's UI tests currently run on Server 2022 VM images. However, the production app is primarily developed for Windows 11. This change moves the UI test jobs to run on Windows 11, so we can run our tests in an environment that's closer to what real users will experience.

### Description of the changes:
- Switched UI tests to target the EssentialExperiences-Win11 and EssentialExperiencesOpenSource-Win11 pools
- Add a step to disable animation effects on the machines. Windows Server has this setting off by default, but Windows 11 has it on by default. Our tests currently don't pass when this is on.
- Target 22000 as the minimum Windows version in all configurations, instead of using a lower minimum version for CI builds.
- Run UI tests in the release pipeline again.
- Retarget the UI test projects to .NET 6
- Rewrite the AoT_ErrorMessage_MessageRetentionUponExitingAoT test so it no longer relies on the invert button to be visible
- <s>Disable the AoT_ErrorMessage_CannotDivideByZero test until we have a way to increase the screen resolution on the machines. We can't use the existing PowerShell cmdlet since it isn't present on Windows desktop editions, and an alternative solution is not ready yet. Re-enabling this test is tracked with #1865</s> In the latest iteration, the ScreenResolutionUtility task is added to increase the resolution, for parity with the existing behavior.
- Other minor cleanup in the UI test code, including deleting dead code and using strongly-typed booleans instead of strings.

### How changes were validated:
- Pipeline run.

### CODE REVIEW

File: build/scripts/TurnOffAnimationEffects.ps1 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    $asTask = $asTaskGeneric.MakeGenericMethod($ResultType)
+    $task = $asTask.Invoke($null, @($WinRtTask))
+    $task.GetAwaiter().GetResult()
+}
+
+[Windows.UI.ViewManagement.Core.UISettingsController,Windows.UI.ViewManagement.Core,ContentType=WindowsRuntime] | Out-Null
+
+$controller = WaitForAsyncAction ([Windows.UI.ViewManagement.Core.UISettingsController]::RequestDefaultAsync())  ([Windows.UI.ViewManagement.Core.UISettingsController])
+
+$controller.SetAnimationsEnabled($false)
```
Comment (tian-lt) [inline build/scripts/TurnOffAnimationEffects.ps1:22]: nit: an empty line here should be better.

**Comment by shweaver-MSFT** [comment]
Changes look good as far as I can tell. But I saw a few errors when I ran the suite of UI tests locally. This is my first time running the project, so I admit I'm not sure if that is expected or not. 

What do you think?

image

**Comment by mcooley** [comment]
@shweaver-MSFT if the currency converter tests are failing, selecting [the right runsettings file](https://github.com/microsoft/calculator/blob/main/src/CalculatorUITests/CalculatorUITests.ci.runsettings) might help. AoT_Tooltip I'm not sure without seeing more of the failure. Note that these tests are passing in the official pipeline.

**Comment by shweaver-MSFT** [comment]
> @shweaver-MSFT if the currency converter tests are failing, selecting [the right runsettings file](https://github.com/microsoft/calculator/blob/main/src/CalculatorUITests/CalculatorUITests.ci.runsettings) might help. AoT_Tooltip I'm not sure without seeing more of the failure. Note that these tests are passing in the official pipeline.

Okay, so any failures I'm seeing are just a side-effect of my personal environment? Gotcha :)

**Review by shweaver-MSFT — APPROVED** [review]

**Review by tian-lt — APPROVED** [review]

**Pull request merged**